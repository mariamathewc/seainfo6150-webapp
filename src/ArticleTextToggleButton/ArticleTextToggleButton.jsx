import React from "react";
import styles from "./ArticleTextToggleButton.module.css";

const ArticleTextToggleButton = (props) => {

    const { buttonText, onClick, slug } = props
   

    return (
        <div >
            <label >
                <button
                    className={styles.btn}
                    id={slug}
                    onClick={(e) => onClick(e, buttonText)}
                >
                    {buttonText}
                </button>
            </label>
        </div>

    );
};

export default ArticleTextToggleButton;