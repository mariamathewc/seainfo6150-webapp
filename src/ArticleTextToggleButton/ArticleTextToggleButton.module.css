.btn {
    border: 3px solid rgba(0,0,0,.29);
    background-color: red;
    font-size: 16px;
    width: 125px;
    position: relative;
    color:white;
}
