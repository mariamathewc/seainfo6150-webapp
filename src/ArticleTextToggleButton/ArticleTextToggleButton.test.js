import React from "react";
import ArticleTextToggleButton from "./ArticleTextToggleButton";
import { fireEvent } from "@testing-library/react";



describe("ArticleTextToggleButton tests", () => {



    it("renders correctly after button is clicked", () => {
        
        const { container, getByText } = render(
            <ArticleTextToggleButton buttonText="Show less" onClick={jest.fn()}  />
        );
        fireEvent.click(getByText("Show less"));
        expect(container).toMatchSnapshot();
    });
 
});