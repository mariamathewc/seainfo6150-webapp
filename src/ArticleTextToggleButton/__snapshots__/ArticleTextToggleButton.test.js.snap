// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleTextToggleButton tests renders correctly after button is clicked 1`] = `
<div>
  <button
    class="button"
  >
    Show less
  </button>
</div>
`;
